backend/app/services/knowledge/code_wiki/navigation.py

```
@@ -0,0 +1,119 @@
+
+
+def _declared_order(knowledge_base: Kind) -> dict[str, int]:
+    spec = (knowledge_base.json or {}).get("spec", {})
+    declared = spec.get(PAGE_ORDER_KEY) or []
+    return {collation_key(str(path)): index for index, path in enumerate(declared)}
+
+
+def _nest(by_key: dict[str, PageNode], order: dict[str, int]) -> list[PageNode]:
+    """Attach each node to its parent, and sort every level the same way."""
+    roots: list[PageNode] = []
+    for key, node in by_key.items():
```

Ruff (0.16.0)

[warning] Loop control variable `key` not used within loop body

Rename unused `key` to `_key`

(B007)